src/hotspot/share/gc/g1/g1FullGCCompactTask.cpp

```diff
@@ -116,6 +117,15 @@ void G1FullGCCompactTask::compact_region(HeapRegion* hr) {
   hr->reset_compacted_after_full_gc(_collector->compaction_top(hr));
 }
 
+G1FullGCCompactTask::G1FullGCCompactTask(G1FullCollector* collector) :
+  G1FullGCTask("G1 Compact Task", collector),
+  _collector(collector),
+  _claimer(collector->workers()) {
+  // Need cleared claim bits for the next concurrent marking.
+  ClassLoaderDataGraph::clear_claimed_marks();
```

Comment:
Is there a good reason to do the clearing here instead of in `void G1FullCollector::complete_collection()`?